clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -286,6 +286,31 @@ void Sema::checkSYCLDeviceVarDecl(VarDecl *Var) {
   checkSYCLType(*this, Ty, Loc, Visited);
 }
 
+// Tests whether given function is a lambda function or '()' operator used as
+// SYCL kernel body function (e.g. in parallel_for).
+//
+// TODO this test should involve checking the caller and matching function types
```

Comment:
Put this is in the TODO document.

Reply:
I will.

JFYI, there are tons of other TODOs in this file which are neither ESIMD-specific, which are not in any TODO file.